examples/README.md

```diff
@@ -0,0 +1,7 @@
+# RMM C++ Examples
+
+This folder contains examples to demonstrate librmm use cases. Running `build.sh` builds all examples.
+
```

Comment:
Tiny nit: To my mind, `build.sh` is slightly more magic than simple didactic examples warrant. Particularly, I would be worried that people just blindly copy it. I _think_ (as shown in the readme for the example itself) that one only really needs some (simpler) cmake incantations.

Reply:
Unfortunately much of the complexity in `build.sh` is necessary to make packaging the examples possible. That's how we ensure they build in CI. I think I want to stick with what we have here, since it matches what cudf does.